src/profiler/profiler.h

```diff
@@ -455,7 +455,7 @@ class Profiler {
   /*! \brief filename to output profile file */
   std::string filename_ = "profile.json";
   /*! \brief profile statistics consist of multiple device statistics */
-  DeviceStats* profile_stat;
+  std::unique_ptr<DeviceStats[], std::default_delete<DeviceStats[]>> profile_stat;
```

Comment:
`std::default_delete<DeviceStats[]>` is default and not needed explicitly, I think